hazelcast/src/test/java/com/hazelcast/jet/core/TestProcessors.java

```diff
@@ -249,14 +296,38 @@ public MockPS setCloseError(Throwable closeError) {
             return this;
         }
 
+        public MockPS initBlocks() {
+            this.initBlocks = true;
+            return this;
+        }
+
+        public MockPS closeBlocks() {
+            this.closeBlocks = true;
+            return this;
+        }
+
+        public static void unblock() {
+            blockingSemaphore.release();
+        }
+
         @Override
-        public void init(@Nonnull Context context) {
+        public boolean initIsCooperative() {
+            return !initBlocks;
+        }
+
+        @Override
+        public void init(@Nonnull Context context) throws InterruptedException {
+            System.out.println("MockPS.init called on " + Thread.currentThread().getName());
             initCalled = true;
             initCount.incrementAndGet();
 
             if (initError != null) {
                 throw sneakyThrow(initError);
             }
+
+            if (initBlocks) {
+                blockingSemaphore.acquire();
```

Comment:
Let's add something like `Thread.sleep(new Random().nextInt(500));` after all `blockingSemaphore.acquire()`, to make the blocking more unpredictable.